### DISCUSSION TITLE
How can we use promotions in argocd along with manual judgement.

### DISCUSSION THREAD

**Discussion by chowmean** [argoproj/argo-cd#6161] [General]
We want to implement argocd with promotions and manual judgment. Since they are not provided directly, how are people doing this. Anyone implemented anything like this. 

Say for below

stage->[ask for promotion]->prod

We have a idea that we can use pre and post sync to check if the deployment is done in stage before proceeding to prod. 

We can also offload this to gitlab. In stages we can apply the deployment first to stage and then to prod. But here the problem is access control. We want few people to have deploy in stage and few can deploy in prod.

**Comment by joebowbeer** [comment]
Can you represent promotion as a PR to a deploy repo?

PR's can generally be configured to require a set of checks (such as post sync status) as well as one or more manual approvals.
